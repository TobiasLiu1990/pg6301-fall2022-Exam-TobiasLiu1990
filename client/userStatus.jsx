import * as React from "react";
import { useState } from "react";
import { Link, Route, Routes, useNavigate } from "react-router-dom";
import { fetchJSON } from "./fetchJson";

export function LoginLinks() {
  return (
    <div>
      <Link to={"/login"}>Login</Link>
      <br></br>
      <Link to={"/register"}>Register new account</Link>
    </div>
  );
}

export function Login() {
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");
  const navigate = useNavigate();

  async function handleSubmit(e) {
    e.preventDefault();
    //POST Login
    const res = await fetch("/api/login", {
      method: "post",
      body: JSON.stringify({ username, password }),
      headers: {
        "content-type": "application/json",
      },
    });
    if (res.ok) {
      navigate("/");
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <h1>Please login!</h1>
      <div>
        Username:{" "}
        <input
          type="text"
          value={username}
          onChange={(e) => setUsername(e.target.value)}
        />
      </div>
      <div>
        Password:{" "}
        <input
          type="password"
          value={password}
          onChange={(e) => setPassword(e.target.value)}
        />
      </div>
      <button>Log in</button>
    </form>
  );
}

export function RegisterNewAccount() {
    const [username, setUsername] = useState("");
    const [password, setPassword] = useState("");
    const [fullName, setFullName] = useState("");
    const navigate = useNavigate();

    async function handleSubmit(e) {
        e.preventDefault();
        navigate("../");

        await fetchJSON("/api/login/register", {
            method: "post",
            json: ({ username, password, fullName }),
        });
    }

    return (
        <form onSubmit={handleSubmit}>
            <h1>Register new user account</h1>
            <div>
                Username:{" "}
                <input
                    type="text"
                    value={username}
                    onChange={(e) => setUsername(e.target.value)}
                />
            </div>
            <div>
                Full name:{" "}
                <input
                    type="text"
                    value={fullName}
                    onChange={(e) => setFullName(e.target.value)}
                />
            </div>
            <div>
                Password:{" "}
                <input
                    type="password"
                    value={password}
                    onChange={(e) => setPassword(e.target.value)}
                />
            </div>
            <button>Log in</button>
        </form>
    );
}

export function LogoutButton(reloadPage) {
    return (
        <button
            onClick={async () => {
                await fetch("/api/login", {
                    method: "delete",
                });
                reloadPage();
            }}
        >
            Log out
        </button>
    );
}

export function UserStatus() {
    return (
        <Routes>
            <Route path={"/"} element={<LoginLinks />} />
            <Route path={"/login"} element={<Login />}></Route>
        </Routes>
    );
}


