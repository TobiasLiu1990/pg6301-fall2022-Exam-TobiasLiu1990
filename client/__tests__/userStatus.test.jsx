import * as React from "react";
import ReactDOM from "react-dom";
import {LoginLinks, LogoutButton} from "../userStatus";
import { MemoryRouter } from "react-router-dom";

describe("userStatus components", () => {
  it("should render LoginLinks", () => {
    const element = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <LoginLinks />
      </MemoryRouter>,
      element
    );

    expect(element.innerHTML).toMatchSnapshot();
  });

  it("should render LogoutButton", () => {
    const element = document.createElement("div");
    ReactDOM.render(<LogoutButton/>, element);

    expect(element.innerHTML).toMatchSnapshot();
  })
});