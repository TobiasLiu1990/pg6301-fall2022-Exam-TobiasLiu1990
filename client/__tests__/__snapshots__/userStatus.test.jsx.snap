// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userStatus components should render LoginLinks 1`] = `"<div><a href="/login">Login</a><br><a href="/register">Register new account</a></div>"`;

exports[`userStatus components should render LogoutButton 1`] = `"<button>Log out</button>"`;
